src/main/windows/editor.js

```diff
@@ -114,94 +149,294 @@ class EditorWindow extends BaseWindow {
 
     // The window is now destroyed.
     win.on('closed', () => {
+      this.lifecycle = WindowLifecycle.QUITTING
```

Comment:
I am not sure, just have a question, does the window is quitting or quitted when the `closed` event is fired. Do we really need `lifecycle` property?

Reply:
Yes, I'll rename it to quitted. We need this state to ensure that we don't access methods of this instance any more. The browser window is destroyed but the class may be used for a short time after the window is destroyed to unload/unregistered events. E.g. async functions.

Reply:
Pls add some comments about this.